From b4d01330c66cbab3563c58f66f73f55726c09aec Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Tue, 5 Apr 2016 17:54:14 +0300
Subject: [PATCH 211/226] staging: fsl-dpaa2/mac: Don't call devm_free_irq

MAC interrupts are registered with devm_request_threaded_irq(), so
there's no need to explicitly unregister them in case of a probe
error or at device remove, as the kernel will take care of that for us.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
(cherry picked from commit 58e0fd23ade4b13e0a3c7e5f201802013e12df1c)
(Stuart: resolved merge conflict)
Signed-off-by: Stuart Yoder <stuart.yoder@nxp.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -429,13 +429,11 @@ static int setup_irqs(struct fsl_mc_devi
 				   DPMAC_IRQ_INDEX, 1);
 	if (err) {
 		dev_err(&mc_dev->dev, "dpmac_set_irq_enable err %d\n", err);
-		goto unregister_irq;
+		goto free_irq;
 	}
 
 	return 0;
 
-unregister_irq:
-	devm_free_irq(&mc_dev->dev, mc_dev->irqs[0]->msi_desc->irq, &mc_dev->dev);
 free_irq:
 	fsl_mc_free_irqs(mc_dev);
 
@@ -456,7 +454,6 @@ static void teardown_irqs(struct fsl_mc_
 	if (err)
 		dev_err(&mc_dev->dev, "dpmac_set_irq_enable err %d\n", err);
 
-	devm_free_irq(&mc_dev->dev, mc_dev->irqs[0]->msi_desc->irq, &mc_dev->dev);
 	fsl_mc_free_irqs(mc_dev);
 }
 
